src/App.tsx

```diff
@@ -1,24 +1,329 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useEffect,
+  useState,
+  useCallback,
+  // useMemo,
+} from 'react';
+import classNames from 'classnames';
+import './App.scss';
+
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+
+import {
+  getTodos,
+  addTodo,
+  deleteTodo,
+  updateTodo,
+} from './api/todos';
+import { BASE_URL } from './utils/fetchClient';
 
-const USER_ID = 0;
+import { ErrorType } from './types/ErrorType';
+import { ErrorShow } from './components/ErrorShow';
+
+import { TodoList, USER_ID } from './components/TodoList/TodoList';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [addingTodoTitle, setAddingTodoTitle] = useState('');
+  const [errorToShow, setErrorToShow] = useState<ErrorType>('none');
+  const [todosToShow, setTodosToShow] = useState(todos);
+  const [chosenFilter, setChosenFilter] = useState<string>('all');
+  const [creating, setCreating] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [processings, setProcessing] = useState<number[]>([]);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then((todosFromServer: Todo[]) => setTodos([...todosFromServer]));
+  }, [todos]);
```

Comment:
You have specified `todos` as a dependency in the useEffect, which may cause unnecessary re-rendering. Consider removing the dependency.